packages/forklift-console-plugin/src/modules/Providers/utils/components/CertificateUpload/FetchCertificateModal.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { FC, useState } from 'react';
+import { calculateThumbprint, useTlsCertificate } from 'src/modules/Providers/hooks';
+import { useModal } from 'src/modules/Providers/modals';
+import { useForkliftTranslation } from 'src/utils/i18n';
+
+import { Loading } from '@kubev2v/common';
+import { Alert, Button, Modal, ModalVariant } from '@patternfly/react-core';
+
+import { VerifyCertificate } from './VerifyCertificate';
+
+export const FetchCertificateModal: FC<{
+  url: string;
+  existingCert: string;
+  handleSave: (cert: string) => void;
+}> = ({ existingCert, url, handleSave }) => {
+  const { toggleModal } = useModal();
+  const { t } = useForkliftTranslation();
+  const [isTrusted, setIsTrusted] = useState(false);
+  const { loading, fetchError, certError, thumbprint, issuer, validTo, certificate } =
+    useTlsCertificate(url);
+  const success = !loading && !fetchError && !certError;
+  const hasThumbprintChanged =
```

Comment:
`thumbprint` will only change once once cert is loaded from "loading" to "some value", when do you expect the `hasThumbprintChanged`  to be true ?

Reply:
```
  const hasThumbprintChanged = existingCert && success && thumbprint !== calculateThumbprint(existingCert);
```
when existing certificate produces a different output then the freshly fetched certificate

Reply:
when do you have existing certificate ?

the flow is:
a. loading
b. get a certificate
c. ask users to approve

where in the flow you can get a change ?